src/hotspot/cpu/s390/gc/shared/barrierSetAssembler_s390.cpp

```diff
@@ -105,16 +105,60 @@ void BarrierSetAssembler::store_at(MacroAssembler* masm, DecoratorSet decorators
   }
 }
 
+// Generic implementation. GCs can provide an optimized one.
```

Comment:
You may want to implement an optimized `G1BarrierSetAssembler::resolve_jobject` and `ModRefBarrierSetAssembler::resolve_jobject`. Otherwise, those GCs may get a regression.

Reply:
I have added implementation for `ModRefBarrierSetAssembler::resolve_jobject`. It seems `G1BarrierSetAssembler::resolve_jobject` is already present, do I need to do anything extra than that ?

Reply:
I think it's fine.